### ISSUE TITLE
--manifest-path provides poor diagnostics

### ISSUE THREAD

**Issue by huonw** [rust-lang/cargo#1772]
e.g. `cargo build --manifest-path some/existing/directory` shows `An unknown error occurred` by default, and `no package found in source` if `--verbose` is passed. It would be nice if at least the latter could be shown by default, but it would also be nice to say something like `manifest-path should point to the Cargo.toml file, not directory`.

Also, `cargo build --manifest-path not/a/real/file` also shows the unknown error by default, and takes `--verbose` to display `No such file or directory (os error 2)`. (It'd be nice for this to mention that the problem is with the `--manifest-path` argument specifically.)

**Comment by carols10cents** [comment]
Since #1955 got merged, I think this is fixed:

```
$ cargo build --manifest-path some/existing/directory
the manifest-path must be a path to a Cargo.toml file
```

```
$ cargo build --manifest-path not/a/real/Cargo.toml
manifest path `not/a/real/Cargo.toml` does not exist
```

Sorry, I should have put a "Fixes #1772" in that issue :)

**Comment by alexcrichton** [comment]
Yay!
